### PR TITLE
[MCA] Dbus-gRPC Migration 

### LINKED ISSUE #23: Combining gRPC(app) with MCA
So far, we have confirmed gRPC, MCA, and AI model working well respectively. Also, I checked that MCA and AI model are working well together. But as we know, gRPC and MCA have some problem(maybe thread safety issue).

If we have any update or get some progress, let's update on this issue page.

### PR DESCRIPTION
**Pull request by sseoreo**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature add
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Testing code
- [ ] Other

## What is the new behavior?
First of all, I change the type of `ModelConnection` from `std::sync::Mutex` to `futures::lock::Mutex`.
The "Send error" is caused by that `std::sync::Mutex` cannot be used in an async function.

## Other information
I'm currently unable to conduct tests. Lately, I've been working on a Mac, but I'm encountering difficulties building the Python model. I've verified that the mca is operational and yielding results from the applications. Therefore, could you please provide guidance on how to build our new onnx model?

### CODE REVIEW

**Comment by hyunsube** [comment]
@sseoreo Okay. I will try on my environment~

File: mca/src/agent/model_connection.rs [hunk 1]
```diff
@@ -18,7 +19,7 @@ unsafe impl Sync for ModelConnection {}
 
 pub trait MCAOperations {
     fn new() -> Self;
-    async fn register_model(&mut self, model_enum: MCAModelEnum) -> zbus::Result<String>;
+    fn register_model(&mut self, model_enum: MCAModelEnum) -> impl std::future::Future<Output = zbus::Result<String>> + Send;
```
Comment (hyunsube) [inline mca/src/agent/model_connection.rs:22]: Is it supposed to work as async function even though we define it as `fn`?

File: mca/src/agent/model_structure.rs [hunk 2]
```diff
@@ -11,19 +11,21 @@ pub struct IrisData {
...
     pub proxy: Model1Proxy<'a>,
 }
 
 // AIModelTrait을 Model1과 Model2에 대한 특성으로 구현
 impl Model1Trait for Model1Struct<'_> {
-    async fn test(&self, input: &str) -> Result<String> {
+    async fn test(&self, input: String) -> Result<String> {
         println!("{:?}", input);
-        let _reply = self.proxy.predict(input).await?;
+        let _reply = self.proxy.predict(&input).await?;
```
Comment (hyunsube) [inline mca/src/agent/model_structure.rs:28]: What is benefit of using `String` instead of `str`?

File: mca/src/api/app_base.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    pub fn new (name: String, addr:String, role: Role) -> Self {
+        Self {
+            appname: name,
+            address: addr,
+            role: role
+        }
+    }
+
+    // Check if the user has permission to perform an action
+    pub fn has_permission(&self, required_role: Role) -> bool {
```
Comment (hyunsube) [inline mca/src/api/app_base.rs:27]: Checking permission! Nice:)

File: mca/src/api/app_manager.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    pub n_apps: u8
+}
+
+impl AppManager {
+    pub fn new() -> Self {
+        AppManager { app_list: HashMap::new(), n_apps:0 }
+    }
+
+    fn register(&mut self, name: String, addr:String, role: Role) -> Arc<Mutex<AppBase>> {
+        let app_point = Arc::new(Mutex::new(AppBase::new(name, addr, role)));
```
Comment (hyunsube) [inline mca/src/api/app_manager.rs:23]: Why `AppBase` is wrapped with `Mutex`? I know `Mutex` is for multi thread environment. Is app planning to use multi thread? Let's discuss at meeting.

**Comment by hyunsube** [comment]
@sseoreo  I just looked over briefly. Let's talk about this at next meeting. Please tell me the role of each file :)

File: mca/src/agent/model_connection.rs (hunk 1 above)
Comment (sseoreo) [reply mca/src/agent/model_connection.rs:22]: Yes, warning message appeared, so I changed it.

File: mca/src/agent/model_structure.rs (hunk 2 above)
Comment (sseoreo) [reply mca/src/agent/model_structure.rs:28]: Actually, it is one of the things I tried to solve the Sync error.
In my opinion, it is not important. it is just different to pass call by reference or call by value, I guess.

File: mca/src/api/app_manager.rs (hunk 4 above)
Comment (sseoreo) [reply mca/src/api/app_manager.rs:23]: This is not used, yet. At first, I planned to assign the struct `AppBase` to each app, and each `AppBase` is referred to in two types of threads:
1. Change Role in AppManager: In this case, `AppBase`(precisely, member `Role`) is written by AppManager
2. Check Author in AppManager: In this case, `AppBase`(precisely, member `Role`) is read by AppManager

I don't think these steps occur at the same time, but I think it should have a mutex. But I cannot sure, haha;

**Pull request merged**